tests/integration_tests/modules/test_power_state_change.py

```diff
@@ -0,0 +1,90 @@
+"""Integration test of the cc_power_state_change module.
+
+Test that the power state config options work as expected.
+"""
+
+import pytest
+import time
+
+from tests.integration_tests.clouds import IntegrationCloud
+from tests.integration_tests.instances import IntegrationInstance
+from tests.integration_tests.log_utils import ordered_items_in_text
+
+USER_DATA = """\
+#cloud-config
+power_state:
+  delay: {delay}
+  mode: {mode}
+  message: msg
+  timeout: {timeout}
+  condition: {condition}
+"""
+
+
+def _detect_reboot(instance: IntegrationInstance):
+    # We'll wait for instance up here, but we don't know if we're
+    # detecting the first boot or second boot, so we also check
+    # the logs to ensure we've booted twice. If the logs show we've
+    # only booted once, wait until we've booted twice
+    instance.instance.wait(raise_on_cloudinit_failure=False)
+    for _ in range(600):
+        try:
+            log = instance.read_from_file('/var/log/cloud-init.log')
+            boot_count = log.count("running 'init-local'")
+            if boot_count == 1:
+                instance.instance.wait(raise_on_cloudinit_failure=False)
+            elif boot_count > 1:
+                break
+        except Exception:
+            pass
+        time.sleep(1)
+    else:
+        raise Exception('Could not detect reboot')
+
+
+def _can_connect(instance):
+    return instance.execute('true').ok
+
+
+# This test is marked unstable because even though it should be able to
+# run anywhere, I can only get it to run in an lxd container, and even then
+# occasionally some timing issues will crop up.
+@pytest.mark.unstable
```

Comment:
It might be nice to have a timed element like we have in the curtin tests: `unstable_until("2021/02/01")` to force us to fix it _at some point_. Not needed for now, though.